lib/broadway_sqs/producer.ex

```diff
@@ -201,6 +209,17 @@ defmodule BroadwaySQS.Producer do
     {:noreply, [], state}
   end
 
+  @impl Producer
+  def prepare_for_draining(%{receive_timer: nil} = state) do
+    {:noreply, [], state}
+  end
+
+  @impl Producer
+  def prepare_for_draining(%{receive_timer: receive_timer} = state) do
+    Process.cancel_timer(receive_timer)
```

Comment:
```suggestion
    receive_timer && Process.cancel_timer(receive_timer)
```

And then we remove the function clause above?